xds/src/main/java/io/grpc/xds/sds/XdsServerBuilder.java

```diff
@@ -119,6 +123,11 @@ public XdsServerBuilder intercept(ServerInterceptor interceptor) {
     return this;
   }
 
+  public XdsServerBuilder tlsContext(DownstreamTlsContext downstreamTlsContext) {
```

Comment:
I think it is okay to allow null - in case the value is passed down from somewhere else and you don't want to special case `null`. In any case this is temporary until the `XdsServerBuilder` is integrated with the XdsClient to get the downstreamTlsContext (even in that case when the downstreamTlsContext could be null or empty).

Reply:
What is the expectation of using Xds client? it should use secure connection? or plaintext is okay?

A proto object cannot be `null` even if the CDS returns empty result. so the caller must be intentionally using `null`, it is not just passed down.

Reply:
If I understand your first question right, and AFAIK the XDS server can specify or allow plaintext for certain connections.

You are right about the `downstreamTlsContext` object never being null in proto - it will just be a default instance. However note that this is just the current/temporary API to use the feature. Once we have integration with XdsClient this API will be removed and  `downstreamTlsContext` will come from LDS (in case of server). At that point I will treat default instance similar to `null`.

Reply:
resolving it for now